### DISCUSSION TITLE
z_index culling of Camera2D

### DISCUSSION THREAD

**Discussion by qt911025** [godotengine/godot-proposals#10786] [2D]
Godot can control visibility by setting "visible" of a node, which is based on tree rather than z_index. I don't know if there is a convenient way to control visibility of nodes above (or below) a specific z_index.
Camera2D is a Node2D, but z_index of Camera2D is useless now. What if Camera2D has a similar feature like frustum culling of 3D camera? Setting z_index of Camera2D can conceal canvas nodes whose z_index is greater than camera's z_index.

**Comment by Mickeon** [comment]
It's not intuitive, and the camera's children are still affected by the `z_index` changing. So if anything, it should be a separate property.
But regardless of how it is done, this _may_ make 2D rendering rather complex for little benefit.
If it's expected that nodes with a certain `z_index` should not be rendered, they could all be inserted inside specific **CanvasLayer** nodes. A **CanvasLayer**'s own `z_index` and `visible` properties could then be changed as needed.
